signer/service/service.go

```diff
@@ -21,3 +41,124 @@ func (s *SignerService) RegisterAPIs(server *oprpc.Server) {
 		Service:   s,
 	})
 }
+
+type SignTransactionResponse struct {
+	Signature hexutil.Bytes `json:"signature"`
+}
+
+func (s *SignerService) SignTransaction(
```

Comment:
What do you think about implementing the RPC `eth_signTransaction` so that existing tooling can "just work" with this service? This RPC is meant to work with a key that is remote from the client

The interface is here:
https://www.quicknode.com/docs/ethereum/eth_signTransaction

The main difference is that you don't pass in the raw transaction, you instead specify the fields. This is slightly less forwards compatible, ie we will manually need to add new transaction types

Reply:
One interesting this about `eth_signTransaction` is that you can specify `from` so then we can manage multiple keys in a single deployment

Reply:
Yeah we are going to support multiple keys, however we are going to select the key based on authenticated client making the request. We can verify it matches the expected from address in the payload, but it is somewhat redundant.

In terms of existing tooling, our authentication for this service will be using mTLS so as long as the the clients can support that then yeah think it would make sense to use this RPC

Reply:
One potential counterpoint to this is that we will eventually use this service to sign the P2P messages the sequencer sends over P2P.

I'm not married to either implementation, but want to ensure that we leave the door open for this functionality in the future.